Externals/expr/include/expr.h

```diff
@@ -468,6 +479,17 @@ extern "C" {
       }
       return i;
     }
```

Comment:
To cleanly extend the lexer, EXPR_TSTRING should be defined and used wherever appropriate (probably everywhere EXPR_TNUMBER is appropriate).

Reply:
I'm not too sure on what this is doing. Do I just put EXPR_TSTRING along side TNUMBER and then check for it when doing the quotations?

Reply:
These flags are used to express which tokens are valid based on the current state of the lexer. Specific kinds of tokens can only appear after certain other tokens.

Option A: Following my original suggestion, you could define EXPR_TSTRING as a unique bit (looks like 1 << 7 is the next unused one), bitwise OR it in all the places TNUMBER is valid, and check for TSTRING in the string case.
Option B: Rename EXPR_TNUMBER to EXPR_TLITERAL. Number and string literals are valid in the same places, so we don't really *need* to distinguish them with a separate bit. Then, we can just check for TLITERAL in the string case.

Thinking about it some more, I'm leaning toward option B, but either works.